### PR TITLE
iOS Referrer fix

### PR DESCRIPTION
**Pull request by ckbo3hrk**
## Type of change

- [x] Bugfix
- [x] Feature
- [x] Code style update (formatting, local variables)

## Description of change

1. Implemented srcdoc ad rendering appoach as suggested in https://github.com/prebid/Prebid.js/issues/977#issue-206471234 for all browsers that support iframe.srcdoc property except  Firefox as it [doesn't send proper http referrer](https://bugzilla.mozilla.org/show_bug.cgi?id=1265961)  for such approach.
2. Fixed url.parse function to initialize pathname property in uniform way for all browsers (with leading /. Current version set pathname without leading / on IE11)

### CODE REVIEW

**Comment by mkendall07** [comment]
@ckbo3hrk 
Can you add a unit test that asserts for the `url.parse` leading / change?

**Comment by ckbo3hrk** [comment]
@mkendall07 I've changed the existing one to cover `url.parse` changed behaviour

File: src/prebid.js [hunk 1]
```diff
@@ -133,6 +133,15 @@ function setRenderSize(doc, width, height) {
   }
 }
 
+/**
+ * Check if parent iframe of passed document supports content rendering via 'srcdoc' property
+ * @param {HTMLDocument} doc document to check support of 'srcdoc'
+ */
+function isSrcdocSupported(doc) {
```
Comment (mkendall07) [inline src/prebid.js]: can you move into `utils.js` please

Comment (ckbo3hrk) [inline src/prebid.js]: done

**Comment by mkendall07** [comment]
LGTM. Thanks!

File: src/prebid.js [hunk 2]
```diff
@@ -299,15 +300,19 @@ $$PREBID_GLOBAL$$.renderAd = function (doc, id) {
...
-        if (doc===document || adObject.mediaType === 'video') {
-          utils.logError('Error trying to write ad. Ad render call ad id ' + id + ' was prevented from writing to the main document.');
+        
+        if (doc === document || adObject.mediaType === 'video') {
+          utils.logError(`Error trying to write ad. Ad render call ad id ${id} was prevented from writing to the main document.`);
         } else if (ad) {
-          doc.write(ad);
-          doc.close();
+          if (isSrcdocSupported(doc)) {
+            doc.defaultView.frameElement.srcdoc = ad;
```
Comment (mkendall07) [inline src/prebid.js]: It looks like `srcdoc` completely wipes out any other content present in the iframe. This is causing DFP to fail to report on viewable impressions. I will be reverting this change until we can find a proper solution.

Comment (mkendall07) [inline src/prebid.js]: @ckbo3hrk fyi

Comment (ckbo3hrk) [inline src/prebid.js]: @mkendall07, I'd like to investigate this bug. Could you provide me a sample dfp tag to reproduce this issue ?

Comment (mkendall07) [inline src/prebid.js]: @ckbo3hrk 
Here is an example:
Working (filter by "activeview" in network)
http://mjacobson.devnxs.net/~mjacobson/discrepancy_testing/test_page_20.html

Not working:
http://mjacobson.devnxs.net/~mjacobson/discrepancy_testing/test_page_21.html

**Pull request merged**